core/state/statedb.go

```diff
@@ -256,6 +256,32 @@ func (self *StateDB) GetState(addr common.Address, hash common.Hash) common.Hash
 	return common.Hash{}
 }
 
+// MerkleProof
+type ProofList [][]byte
+
+func (n *ProofList) Put(key []byte, value []byte) error {
+	*n = append(*n, value)
+	return nil
+}
+
+// returns the MerkleProof for a given Account
+func (self *StateDB) GetProof(a common.Address) [][]byte {
+	var proof ProofList
+	self.trie.Prove(crypto.Keccak256(a.Bytes()), 0, &proof)
+	return proof
+}
+
+// returns the StorageProof for given key
+func (self *StateDB) GetStorageProof(a common.Address, key common.Hash) [][]byte {
+	trie := self.StorageTrie(a)
+	if trie == nil {
+		return [][]byte{}
+	}
+	var proof ProofList
+	trie.Prove(crypto.Keccak256(key.Bytes()), 0, &proof)
```

Comment:
Same here -- better to bubble up errors (and make your own) instead of relying on the sender to explicitly check if returnvalues are empty

Reply:
fixed